File: pdf-ai-summarizer/frontend/types/document.ts
export type DocumentUploadResponse = {
  id: string;
  original_filename: string;
  stored_filename: string;
  file_size: number;
  content_type: string;
};

export type DocumentTextPreviewResponse = {
  id: string;
  page_count: number;
  text_length: number;
  preview: string;
};

File: pdf-ai-summarizer/backend/app/schemas/document.py
from pydantic import BaseModel

class DocumentUploadResponse(BaseModel):
    id: str
    original_filename: str
    stored_filename: str
    file_size: int
    content_type: str

class DocumentTextPreviewResponse(BaseModel):
    id: str
    page_count: int
    text_length: int
    preview: str

class DocxAiConvertRequest(BaseModel):
    extraction_mode: str = "text"

class MergePdfRequest(BaseModel):
    document_ids: list[str]